src/Sniff/PunctuationSpacingSniff.php

```diff
@@ -11,21 +11,58 @@
  */
 final class PunctuationSpacingSniff extends AbstractSpacingSniff
 {
-    private const NO_SPACE_BEFORE = [')', ']', '}', ':', '.', ',', '|'];
-    private const NO_SPACE_AFTER = ['(', '[', '{', '.', '|'];
-    private const ONE_SPACE_AFTER = [':', ','];
+    private const DEFAULT_CONFIG_BEFORE = [
+        ')' => 0,
+        ']' => 0,
+        '}' => 0,
+        ':' => 0,
+        '.' => 0,
+        ',' => 0,
+        '|' => 0,
+    ];
+    private const DEFAULT_CONFIG_AFTER = [
+        '(' => 0,
+        '[' => 0,
+        '{' => 0,
+        '.' => 0,
+        '|' => 0,
+        ':' => 1,
+        ',' => 1,
+    ];
+
+    /**
+     * @var array<string, int|null>
+     */
+    private array $spaceBeforeConfig;
+
+    /**
+     * @var array<string, int|null>
+     */
+    private array $spaceAfterConfig;
+
+    /**
+     * @param array<string, int|null> $spaceBeforeOverride
+     * @param array<string, int|null> $spaceAfterOverride
+     */
+    public function __construct(
+        array $spaceBeforeOverride = [],
+        array $spaceAfterOverride = []
+    ) {
+        $this->spaceBeforeConfig = array_merge(self::DEFAULT_CONFIG_BEFORE, $spaceBeforeOverride);
+        $this->spaceAfterConfig = array_merge(self::DEFAULT_CONFIG_AFTER, $spaceAfterOverride);
+    }
 
     /**
      * @param list<Token> $tokens
      */
     protected function shouldHaveSpaceBefore(int $tokenPosition, array $tokens): ?int
```

Comment:
Given this method's name I would expect a boolean as return value.

Reply:
Indeed

Reply:
WDYT about `getExpectedSpacesAfter` and `getExpectedSpaceBefore` @JoolsMcFly ? Something else ?

Reply:
Or simply `getSpacesAfter` and `getSpacesBefore`?